### ISSUE TITLE
Identify network analysis functions that don't use the C core

### ISSUE THREAD

**Issue by szhorvat** [igraph/rigraph#1415]
Write a list of network analysis functions that don't use the C core (such as `subgraph_centrality()`).

@krlmlr Can you guys help with this, so we can evaluate which of these should be moved to the C core?

----

### List of functions

 - [ ] `subgraph_centrality()`
 - [ ] `power_centrality()`
 - [ ] `alpha_centrality()`
 - [ ] `match_vertices()`
 - [ ] `layout_nicely()` — the heuristics for selecting an appropriate layout could be moved to the C core
 - [ ] ... ?

**Comment by krlmlr** [comment]
Would that be all functions that don't use `*_impl()` or `.Call()`, directly or indirectly?

I'm having issues with the flow package in this context, https://github.com/moodymudskipper/flow/issues/175 .

**Comment by szhorvat** [comment]
> Would that be all functions that don't use `*_impl()` or `.Call()`, directly or indirectly?

Yes, that's right, but I'm only interested in network analysis functions, i.e. those which could in principle be implemented in C.

I know of at least `power_centrality()`, `subgraph_centrality()`, `alpha_centrality()`.

**Comment by krlmlr** [comment]
Can you restrict what functions you are looking for in this list?

https://r.igraph.org/reference/index.html

**Comment by szhorvat** [comment]
@krlmlr Let's not do a full survey. If you come across something that you think might benefit from moving it to the C core, please add it to the list at the top.

**Comment by schochastics** [comment]
Claude did not find any other relevant functions besides the one mentioned on top

**Comment by maelle** [comment]
So what are the TODOs? Open issues in the C core repo for those functions?

**Comment by krlmlr** [comment]
@szhorvat: Do you still think it's a good idea to create C implementations for these functions? Should we try an automatic translation and perhaps prototype here?
